sources/packages/backend/apps/load-test-gateway/src/services/workflow/workflow-data-load.service.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  Application,
+  EducationProgramOffering,
+  Student,
+  StudentAssessment,
+  StudentAssessmentStatus,
+  User,
+} from "@sims/sims-db";
+import {
+  createFakeApplication,
+  createFakeEducationProgramOffering,
+  createFakeStudent,
+  createFakeStudentAssessment,
+  createFakeUser,
+} from "@sims/test-utils";
+import { DataSource, Repository } from "typeorm";
+import { APPLICATION_DATA_SINGLE_INDEPENDENT } from "../../utils/application.constants";
+
+interface ApplicationPreliminaryData {
+  offering: EducationProgramOffering;
+  student: Student;
+}
+
+@Injectable()
+export class WorkflowDataLoadService {
+  constructor(
+    private readonly dataSource: DataSource,
+    @InjectRepository(Application)
+    private readonly applicationRepo: Repository<Application>,
+    @InjectRepository(EducationProgramOffering)
+    private readonly offeringRepo: Repository<EducationProgramOffering>,
+    @InjectRepository(Student)
+    private readonly studentRepo: Repository<Student>,
+    @InjectRepository(User)
+    private readonly userRepo: Repository<User>,
+    @InjectRepository(StudentAssessment)
+    private readonly assessmentRepo: Repository<StudentAssessment>,
+  ) {}
+
+  /**
+   * Load the application and assessment data required for the load test.
+   * @param dataIterations load test iterations.
+   * @returns student assessments.
+   */
+  async loadApplicationAssessmentData(
+    dataIterations: number,
+  ): Promise<StudentAssessment[]> {
+    const preliminaryData = await this.createApplicationPreliminaryData();
+    const applications: Application[] = [];
+    for (let i = 1; i <= dataIterations; i++) {
+      const submittedApplicationData = {
+        ...APPLICATION_DATA_SINGLE_INDEPENDENT,
+        selectedOffering: preliminaryData.offering.id,
+      };
+      const application = createFakeApplication(
+        {
+          student: preliminaryData.student,
+          location: preliminaryData.offering.institutionLocation,
+        },
+        { initialValue: { data: submittedApplicationData } },
+      );
+      applications.push(application);
+    }
+    return this.saveApplicationAssessments(
+      applications,
+      preliminaryData.offering,
+    );
+  }
+
+  /**
+   * Create preliminary data required to build the iteration of applications.
+   * @returns preliminary data.
+   */
+  private async createApplicationPreliminaryData(): Promise<ApplicationPreliminaryData> {
+    const auditUser = await this.userRepo.save(createFakeUser());
+    const [offering, student] = await Promise.all([
+      this.offeringRepo.save(createFakeEducationProgramOffering({ auditUser })),
+      this.studentRepo.save(createFakeStudent()),
+    ]);
+    return { offering, student };
+  }
+
+  /**
+   * Save the application and assessment data.
+   * @param applications applications.
+   * @param offering offering.
+   * @returns assessments.
+   */
+  private async saveApplicationAssessments(
+    applications: Application[],
```

Comment:
Similar question from https://github.com/bcgov/SIMS/pull/2477/files#r1384156405
Maybe we can have a sync-up to solidify the intentions.